samples/client/petstore/java/webclient-useSingleRequestParameter/src/main/java/org/openapitools/client/api/PetApi.java

```diff
@@ -585,7 +585,7 @@ public ResponseSpec updatePetWithResponseSpec(Pet pet) throws WebClientResponseE
         return updatePetRequestCreation(pet);
     }
 
-    public class UpdatePetWithFormRequest {
+    public static class UpdatePetWithFormRequest {
```

Comment:
Thanks for the PR.

It looks like this is a breaking change, right?

If I've the following:

```java
PetApi.UpdatePetWithFormRequest a = new PetApi().new UpdatePetWithFormRequest(12L, "ab", "bc");
```

I'll get an error after `UpdatPetWithFormRequest` becomes static.

Reply:
Yes, that's correct, it has to be updated to:

```java
PetApi.UpdatePetWithFormRequest a = new PetApi.UpdatePetWithFormRequest(12L, "ab", "bc");
```

Reply:
then i think we need an add an option for fallback as the upcoming v7.12.0 release only allows breaking changes with fallback only.

Reply:
Sure, any pointers/docs/example for the fallback strategy?

Reply:
Please have a look at https://github.com/OpenAPITools/openapi-generator/pull/18072 (for c client generator) as it adds an option to decide whether the output should be JSON-formatted 

Commit https://github.com/OpenAPITools/openapi-generator/pull/18072/commits/424e3cccf1670f257403e85aa3f49160a723c44a in the PR adds the option

Reply:
actually please put it on hold....

Reply:
i think another way is to update the existing option `useSingleRequestParameter` to allow `static` instead of just true or false so that we don't need to add another option

Reply:
can you please take a look at updating `useSingleRequestParameter` to allow `static`?

(when static is provide by the user, we will add "staticRequest" to true in `additionalProperties` so that {{#staticRequest}} ... {{/staticRequest}} can be used in the template to add `static` in the output

Reply:
I'll dig into that, thanks for the pointers

Reply:
Added initial code using the `static` value, take a look, if the overall implementation is sufficient I'm happy to refactor to polish details. Thanks @wing328

Reply:
looks good